### ISSUE TITLE
Image Size selector missing

### ISSUE THREAD

**Issue by mcampal** [Automattic/wp-calypso#47302]
#### Steps to reproduce

1. Upload a picture to a Simple site
2. Upgrade it to a Business plan and activate Karuna theme
3. Create an Image block with the uploaded picture

#### What I expected

Available ‘Image Size’ selector on the Image Block properties when editing a post or page

#### What happened instead

Only Image dimensions (width and height) fields are displayed but no dropdown image size selector 

#### Browser / OS version

Chrome / Windows 10

#### Screenshot / Video
This capture shows a ‘pre-upgrade’ uploaded image on the user’s site:
![ok_image_size_full_ok](https://user-images.githubusercontent.com/72819905/98744434-76c7cb00-2390-11eb-87ac-6ccd179a8805.png)

This capture shows a ‘post-upgrade’ uploaded image on the user’s site:
![no_image_size_full_ok](https://user-images.githubusercontent.com/72819905/98744428-74657100-2390-11eb-8ae6-ab6ed929a438.png)

#### Context / Source

Please note that the dropdown image size selector comes back after reverting the site to a simple site.

**Comment by cathymcbride** [comment]
Some additional discussion here: p9F6qB-64S-p2

**Comment by jeyip** [comment]
This is a curious issue. I can verify the issue on the affected user's site, but I can't seem to reproduce it on a testing site. 

- I create a new simple site through https://wordpress.com/new
@mcampal maybe you have insight into wh

**Comment by galakhyati** [comment]
The Image Size option is not showing for the images that were existing on the page/post while the site was simple: https://d.pr/v/HF7WVg/dUSpizYvO8

So the only missing steps from the ones you followed was to add an image on a page/post while the site was on a simple site, not just uploaded to the media.

Now if you add another image on the atomic site, it shows the Image Size option: https://d.pr/v/RZEspI/4qP7G55OMf

So basically the existing images on an atomic site is not showing the Image Size option. Hope that helps, please ping me if need any further help in replicating the issue.

**Comment by tjcafferkey** [comment]
I am able to replicate this issue by

* Creating a page on a Simple Site and inserting an image > Save page
* Upgrade Simple Site to Atomic
* Edit the page, click the image that was inserted when your site was a Simple Site.

I suspect it has something to do with the transfer of data from Simple to Atomic sites, I found an article which explains more about this in the link at the bottom of this comment.

I can see that some of the settings on the image are missing after the site gets upgraded to Atomic. You can see in the below screenshot that after the site becomes Atomic, the image settings loses `sizes` which existed prior to the transfer. My knowledge falls short on where this loss of data actually happens.

![Screenshot 2021-01-20 at 10 44 52](https://user-images.githubusercontent.com/8639742/105164204-be936d00-5b0c-11eb-820f-7d18768221ff.png)

**Some resources for anyone else looking into this:**
* Debugging Atomic Gutenberg locally: PCYsg-t5D-p2 (Skip to "Develop locally with Requestly")
* Simple > Atomic site creation process: PCYsg-uAn-p2

**Comment by edwinho89** [comment]
36074230-hc
